lib/aes_256_cbc_encryptor.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+class Aes256CbcEncryptor
+  def initialize(hex_secret, hex_iv)
+    @secret = [hex_secret].pack('H*')
+    @iv = [hex_iv].pack('H*')
+    raise ArgumentError, 'Secret must be 32 bytes.' unless @secret.length == 32
+    raise ArgumentError, 'IV must be 16 bytes.' unless @iv.length == 16
+  end
+
+  attr_reader :secret, :iv
+
+  def encrypt(payload)
+    cipher = _encryption_cipher
```

Comment:
what's the reasoning for the prefixed underscore on these method names? Was this moving toward some sort of memoization or is this just semi-pythonic?

Reply:
hahaha, it's a private method, i've seen some ruby libraries do it that way. In fact, I think we don't want these to be memoized otherwise it could cause issues. I was planning on DRYing things up a bit but after doing a few tests in the console I thought it was safer to just be explicit about the encryptor and decryptor. You basically have to set the type of action you want to perform on the cipher, so there isn't a really good way to memoize. I can do a little bit more experimentation to find out if there is a better way to do it, but this worked so i stuck with it.